test/bh/mix-test.js

```diff
@@ -0,0 +1,132 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('mix()(×)', function() {
+  it('should return mix', function() {
+    compile(function() {
+      block('button').def()(function() {
+        return JSON.stringify(this.ctx.mix);
+      });
+    })
+      .apply({ block: 'button', mix: { block: 'mix' } })
+      .should.equal('{"block":"mix"}');
+  });
+
+  it('should set single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix' })
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set single mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return { block: 'mix' }; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix', function() {
+    compile(function() {
+      block('button').mix()([ { block: 'mix' } ]);
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return [ { block: 'mix' } ]; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  xit('should extend user single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix2' });
+    })
+      .apply({ block: 'button', mix: { block: 'mix1' } })
+      .should.equal('<div class="button mix1 mix2"></div>');
+  });
+
+  xit('should extend user array mix', function() {
+    compile(function() {
+      block('button')(
+        mix()([ { block: 'mix' } ])
+      );
+    })
+      .apply({ block: 'button', mix: [ { block: 'user-mix' } ] })
+      .should.equal('<div class="button user-mix mix"></div>');
+  });
+
+  xit('should extend later declarations', function() {
```

Comment:
Can we think about integrating https://github.com/bem/bem-core/issues/820 (btw should we move it here?).

Something like (with es6 and #171):
```js
block('button').mix()(node => node.addMixTo({block: 'b'});
```